core/network/protocols/propagate_transactions_protocol.cpp

```diff
@@ -300,34 +310,39 @@ namespace kagome::network {
     });
   }
 
-  //  void PropagateTransactionsProtocol::writePropagatedExtrinsics(
-  //      std::shared_ptr<Stream> stream,
-  //      const PropagatedExtrinsics &message,
-  //      std::function<void(outcome::result<std::shared_ptr<Stream>>)> &&cb) {
-  //    auto read_writer = std::make_shared<ScaleMessageReadWriter>(stream);
-  //
-  //    read_writer->write(message,
-  //                       [stream, wp = weak_from_this(), cb = std::move(cb)](
-  //                           auto &&write_res) mutable {
-  //                         auto self = wp.lock();
-  //                         if (not self) {
-  //                           stream->reset();
-  //                           if (cb) cb(ProtocolError::GONE);
-  //                           return;
-  //                         }
-  //
-  //                         if (not write_res.has_value()) {
-  //                           SL_VERBOSE(
-  //                               self->log_,
-  //                               "Error while writing propagated extrinsics:
-  //                               {}", write_res.error().message());
-  //                           stream->reset();
-  //                           if (cb) cb(write_res.as_failure());
-  //                           return;
-  //                         }
-  //
-  //                         if (cb) cb(stream);
-  //                       });
-  //  }
+  void PropagateTransactionsProtocol::propagateTransactions(
+      gsl::span<const primitives::Transaction> txs) {
+    SL_DEBUG(log_, "Propagate transactions : {} extrinsics", txs.size());
+
+    std::vector<libp2p::peer::PeerId> peers;
+    stream_engine_->forEachPeer(
+        [&peers](const libp2p::peer::PeerId &peer_id, const auto &) {
+          peers.push_back(peer_id);
+        });
+    if (peers.size() > 1) {
+      for (const auto &tx : txs) {
+        if (auto key = ext_event_key_repo_->getEventKey(tx); key.has_value()) {
+          extrinsic_events_engine_->notify(
+              key.value(),
+              primitives::events::ExtrinsicLifecycleEvent::Broadcast(
+                  key.value(), std::move(peers)));
```

Comment:
really questionable peers move
what would be on next for loop iteration?

Reply:
It was copied from gossiper implementation. I'll check it anyway.

Reply:
You right. Fixed in next commit.